internal/quantile/aggregate.go

```diff
@@ -74,36 +74,43 @@ func (e *E2eProcessingLatencyAggregate) Less(i, j int) bool {
 	return e.Percentiles[i]["percentile"] > e.Percentiles[j]["percentile"]
 }
 
+// Add merges two E2eProcessingLatencyAggregate by averaging the percentiles
+//
+// It is side-effect free and does not modify the receiver or the parameter
 func (e *E2eProcessingLatencyAggregate) Add(e2 *E2eProcessingLatencyAggregate, N int) *E2eProcessingLatencyAggregate {
 	if e == nil {
-		*e = *e2
```

Comment:
I didn't either, to be honest.

I'm just trying to be consistent, now.  Before, depending on the branch it took it would either modify the receiver or not.